libraries/adaptive-expressions/src/functionUtils.ts

```diff
@@ -0,0 +1,1033 @@
+
+        FunctionUtils.validateArityAndAnyType(expression, 1, Number.MAX_SAFE_INTEGER, ReturnType.Number);
+    }
+
+    /**
+     * Validate 1 or more string arguments.
+     * @param expression Expression to validate.
+     */
+    public static validateString(expression: any): void {
+        FunctionUtils.validateArityAndAnyType(expression, 1, Number.MAX_SAFE_INTEGER, ReturnType.String);
+    }
+
+    /**
+     * Validate there are two children.
+     * @param expression Expression to validate.
+     */
+    public static validateBinary(expression: any): void {
+        FunctionUtils.validateArityAndAnyType(expression, 2, 2);
+    }
+
+    /**
+     * Validate 2 numeric arguments.
+     * @param expression Expression to validate.
+     */
+    public static validateBinaryNumber(expression: any): void {
+        FunctionUtils.validateArityAndAnyType(expression, 2, 2, ReturnType.Number);
+    }
+
+    /**
+     * Validate 1 or 2 numeric arguments.
+     * @param expression Expression to validate.
+     */
+    public static validateUnaryOrBinaryNumber(expression: any): void {
+        FunctionUtils.validateArityAndAnyType(expression, 1, 2, ReturnType.Number);
+    }
+
+    /**
+     * Validate 2 or more than 2 numeric arguments.
+     * @param expression Expression to validate.
+     */
+    public static validateTwoOrMoreThanTwoNumbers(expression: any): void {
+        FunctionUtils.validateArityAndAnyType(expression, 2, Number.MAX_VALUE, ReturnType.Number);
+    }
+
+    /**
+     * Validate there are 2 numeric or string arguments.
+     * @param expression Expression to validate.
+     */
+    public static validateBinaryNumberOrString(expression: any): void {
+        FunctionUtils.validateArityAndAnyType(expression, 2, 2, ReturnType.Number | ReturnType.String);
+    }
+
+    /**
+     * Validate there is a single argument.
+     * @param expression Expression to validate.
+     */
+    public static validateUnary(expression: any): void {
+        FunctionUtils.validateArityAndAnyType(expression, 1, 1);
+    }
+
+    /**
+     * Validate there is a single argument.
+     * @param expression Expression to validate.
+     */
+    public static validateUnaryNumber(expression: any): void {
+        FunctionUtils.validateArityAndAnyType(expression, 1, 1, ReturnType.Number);
+    }
+
+    /**
+     * Validate there is a single string argument.
+     * @param expression Expression to validate.
+     */
+    public static validateUnaryString(expression: any): void {
+        FunctionUtils.validateArityAndAnyType(expression, 1, 1, ReturnType.String);
+    }
+
+    /**
+     * Validate there is a single boolean argument.
+     * @param expression Expression to validate.
+     */
+    public static validateUnaryBoolean(expression: any): void {
+        FunctionUtils.validateOrder(expression, undefined, ReturnType.Boolean);
+    }
+
+    /**
+     * Verify value is numeric.
+     * @param value alue to check.
```

Comment:
>alue [](start = 20, length = 4)

Value

Reply:
These still need to be initial cap.

---
In reply to: [454697396](https://github.com/microsoft/botbuilder-js/pull/2502#discussion_r454697396) [](ancestors = 454697396)